pkg/iplister/iplister.go

```diff
@@ -0,0 +1,67 @@
+package iplister
+
+import (
+	"context"
+	"errors"
+	"net"
+	"time"
+)
+
+const (
+	defaultTimeout = time.Minute
+)
+
+type IPLister struct {
+	reader  Reader
+	decoder Decoder
+	timeout time.Duration
+}
+
+func New(reader Reader, decoder Decoder, opts ...iplisterOption) *IPLister {
+	i := &IPLister{
+		reader:  reader,
+		decoder: decoder,
+		timeout: defaultTimeout,
+	}
+
+	for _, opt := range opts {
+		opt(i)
+	}
+
+	return i
+}
+
+func (i *IPLister) GetIPs() ([]string, error) {
```

Comment:
Should the implementor get a chance to pass in their ctx?
```golang
func (i *IPLister) GetIPs(ctx context.Context) ([]string, error) {
```

Reply:
@whwalter can we modify the interface contract?

Reply:
I didn't see an interface defined for an iplister.

Reply:
Oh I was just being cheeky. It's here
https://github.com/kanopy-platform/argoslower/pull/16/files#diff-be71d4a8fb1a187ece7b537b7a24fe2dc96d1e7ed0600fdd8efe7c54000159dfR194-R197

I'll merge this and can update later if needed, easy change

Reply:
That explains why I didn't see it in the codebase.  It was in a branch. sg. it can be changed later if necessary.